textalyze.rb

```diff
@@ -23,82 +23,91 @@
 # In short, item_counts(array) tells us how many times each item appears
 # in the input array.
 
-def item_counts(array)
+require 'open-uri'
+require 'nokogiri'
+
+def item_counts(words_array)
   #counts = {} # Initialize counts to an empty Hash
   counts = Hash.new(0)  #This sets the the value to zero
-
-  array.each do |item|
-    # Add code here to modify the "counts" hash accordingly
-    # You'll need to handle two cases:
-    #   1. The first time we've seen a particular item in the array
-    #   2. The second-or-later time we've seen a particular item in the array
+  #This counts the letters ot words from the array and creates a key value hash
+  words_array.each do |item|
     counts[item] += 1
   end
   return counts # This returns the "counts" hash
 end
 
 def string_split(string)
+  #Might get rid of this method.
   string = sanitize(string)
   string_split = string.split("")
   return string_split
 end
 
-def sanitize(string)
-  sanitized = string.downcase.scan(/[a-z]/)  #trying scan
-  return sanitized
+def words_array(words)
+  #thinking of changing this method name.
+  #letter_sanitized = string.downcase.scan(/[a-z]/)  #trying scan
+  words_array = words.downcase.scan(/\w+/)  #trying scan
+  return words_array  #Returns words as an array. Not sure if I still need letters.
 end
 
 def read_file(filename)
-  begin
-    file_contents = File.read(filename)
-  rescue
-    puts "Error opening file"
-  end
+  #This method reads the file
+  file_contents = File.read(filename)
   return file_contents
```

Comment:
As above, you don't need to assign the return value to a variable before returning, i.e.,

```ruby
def read_file(filename)
  return File.read(filename)
end
```